typespec-tests/src/main/java/com/cadl/patch/models/Salmon.java

```diff
@@ -261,47 +246,33 @@ private JsonWriter toJsonMergePatch(JsonWriter jsonWriter) throws IOException {
     @Generated
     public static Salmon fromJson(JsonReader jsonReader) throws IOException {
         return jsonReader.readObject(reader -> {
-            String id = null;
-            String name = null;
-            int age = 0;
-            String color = null;
-            String kind = "salmon";
-            List<Fish> friends = null;
-            Map<String, Fish> hate = null;
-            Fish partner = null;
+            Salmon deserializedSalmon = new Salmon();
             while (reader.nextToken() != JsonToken.END_OBJECT) {
                 String fieldName = reader.getFieldName();
                 reader.nextToken();
 
                 if ("id".equals(fieldName)) {
-                    id = reader.getString();
+                    JsonMergePatchHelper.getFishAccessor().setId(deserializedSalmon, reader.getString());
                 } else if ("name".equals(fieldName)) {
-                    name = reader.getString();
+                    JsonMergePatchHelper.getFishAccessor().setName(deserializedSalmon, reader.getString());
                 } else if ("age".equals(fieldName)) {
-                    age = reader.getInt();
+                    JsonMergePatchHelper.getFishAccessor().setAge(deserializedSalmon, reader.getInt());
                 } else if ("color".equals(fieldName)) {
-                    color = reader.getString();
+                    JsonMergePatchHelper.getFishAccessor().setColor(deserializedSalmon, reader.getString());
                 } else if ("kind".equals(fieldName)) {
-                    kind = reader.getString();
+                    deserializedSalmon.kind = reader.getString();
                 } else if ("friends".equals(fieldName)) {
-                    friends = reader.readArray(reader1 -> Fish.fromJson(reader1));
+                    List<Fish> friends = reader.readArray(reader1 -> Fish.fromJson(reader1));
+                    deserializedSalmon.friends = friends;
                 } else if ("hate".equals(fieldName)) {
-                    hate = reader.readMap(reader1 -> Fish.fromJson(reader1));
+                    Map<String, Fish> hate = reader.readMap(reader1 -> Fish.fromJson(reader1));
+                    deserializedSalmon.hate = hate;
                 } else if ("partner".equals(fieldName)) {
-                    partner = Fish.fromJson(reader);
+                    deserializedSalmon.partner = Fish.fromJson(reader);
                 } else {
                     reader.skipChildren();
                 }
             }
-            Salmon deserializedSalmon = new Salmon();
-            deserializedSalmon.setId(id);
-            deserializedSalmon.setName(name);
-            deserializedSalmon.setAge(age);
-            deserializedSalmon.setColor(color);
-            deserializedSalmon.kind = kind;
-            deserializedSalmon.friends = friends;
-            deserializedSalmon.hate = hate;
-            deserializedSalmon.partner = partner;
 
             return deserializedSalmon;
```

Comment:
I guess I am late.

One possible (simpler) alternative could be call `updatedProperties.clear()` in the end of `fromJson`, and re-add the `kind` to the dict.

Reply:
That could work as well. I'll continue with the access helper pattern though as we may need this elsewhere and this extends the good baseline @haolingdong-msft set up.